generate-wrapper.py

```diff
@@ -23,25 +23,67 @@ def typename(decl):
+        "double": "number",
+        "char*": "string",
+        "char**": "out_string_wrapper",
+        "const char*": "string",
+        "const char**": "out_string_wrapper",
+        "float": "number",
+        "idx_t": "number",
+        "idx_t*": "idx_pointer",
+        "int8_t": "number",
+        "int16_t": "number",
+        "int32_t": "number",
+        "int64_t": "number", # should use bigint because max safe int in JS is 2^53-1
+        "uint8_t": "number",
+        "uint16_t": "number",
+        "uint32_t": "number",
+        "uint64_t": "number", # should use bigint because max safe int in JS is 2^53-1
+        "uint64_t*": "uint64_pointer",
+        "size_t": "number",
+        "void": "void",
+        "void*": "pointer",
+        "const void*": "pointer",
+    }
 
     def visit_TypeDecl(self, node):
         name = node.declname
         if not name.startswith('duckdb_'):
             return
 
         if isinstance(node.type, pycparser.c_ast.Struct):
-            self.result += f'exports.Set(Napi::String::New(env, "{name}"), duckdb_node::PointerHolder<{name}>::Init(env, "{name}")->Value());\n'
-
-        if isinstance(node.type, pycparser.c_ast.Enum):
-            self.result += f'auto {name}_enum = Napi::Object::New(env);\n'
+            self.cpp_result += f'exports.Set(Napi::String::New(env, "{name}"), duckdb_node::PointerHolder<{name}>::Init(env, "{name}")->Value());\n'
+            self.types_result += f'export class {name} {{}}\n'
+            self.c_type_to_ts_type[name] = name
+            self.c_type_to_ts_type[f'{name}*'] = name
+
+        elif isinstance(node.type, pycparser.c_ast.Enum):
+            self.cpp_result += f'auto {name}_enum = Napi::Object::New(env);\n'
+            self.types_result += f'export enum {name} {{\n'
+            self.c_type_to_ts_type[name] = name
             enum_idx = 0
             for enum in node.type.values.enumerators:
                 if enum.value is not None:
                     enum_idx = int(enum.value.value)
-                self.result += f'{name}_enum.Set("{enum.name}", {enum_idx});\n'
+                self.cpp_result += f'{name}_enum.Set("{enum.name}", {enum_idx});\n'
+                self.types_result += f'  {enum.name} = {enum_idx},\n'
                 enum_idx += 1
-            self.result += f'exports.DefineProperty(Napi::PropertyDescriptor::Value("{name}", {name}_enum, static_cast<napi_property_attributes>(napi_enumerable | napi_configurable)));\n'
+            self.cpp_result += f'exports.DefineProperty(Napi::PropertyDescriptor::Value("{name}", {name}_enum, static_cast<napi_property_attributes>(napi_enumerable | napi_configurable)));\n'
+            self.types_result += f'}}\n'
+        
+        elif typename(node.type) == 'void':
+            # Do these void* types need any corresponding Napi code?
+            self.types_result += f'export class {name} {{}}\n'
+            self.c_type_to_ts_type[name] = name
+            self.c_type_to_ts_type[f'{name}*'] = name
```

Comment:
This part handles typedefs like `typedef void *duckdb_bind_info;`. It's needed because some APIs take or return `duckdb_bind_info`s.